UnitTests/GitUI.Tests/UserControls/FilterInfoTests.cs

```diff
@@ -741,5 +741,101 @@ public void FilterInfo_GetRevisionFilter(string author, string committer, string
             filterInfo.GetSummary().Should().Be(expectedSummary);
             filterInfo.GetRevisionFilter().ToString().Should().Be(expectedArgs);
         }
+
+        [Test]
+        public void FilterInfo_Apply_ByCommitMessage()
+        {
+            FilterInfo filterInfo = new();
+            var filterLaunched = filterInfo.Apply(new RevisionFilter("message", true, false, false, false));
+
+            filterLaunched.Should().BeTrue();
+            filterInfo.ByMessage.Should().BeTrue();
+            filterInfo.Message.Should().Be("message");
+
+            filterInfo.ByAuthor.Should().BeFalse();
+            filterInfo.ByCommitter.Should().BeFalse();
+            filterInfo.ByDiffContent.Should().BeFalse();
+        }
+
+        [Test]
+        public void FilterInfo_Apply_ByCommitter()
+        {
+            FilterInfo filterInfo = new();
+            var filterLaunched = filterInfo.Apply(new RevisionFilter("committer", false, true, false, false));
+
+            filterLaunched.Should().BeTrue();
+            filterInfo.ByCommitter.Should().BeTrue();
+            filterInfo.Committer.Should().Be("committer");
+
+            filterInfo.ByMessage.Should().BeFalse();
+            filterInfo.ByAuthor.Should().BeFalse();
+            filterInfo.ByDiffContent.Should().BeFalse();
+        }
+
+        [Test]
+        public void FilterInfo_Apply_ByAuthor()
+        {
+            FilterInfo filterInfo = new();
+            var filterLaunched = filterInfo.Apply(new RevisionFilter("author", false, false, true, false));
+
+            filterLaunched.Should().BeTrue();
+            filterInfo.ByAuthor.Should().BeTrue();
+            filterInfo.Author.Should().Be("author");
+
+            filterInfo.ByMessage.Should().BeFalse();
+            filterInfo.ByCommitter.Should().BeFalse();
+            filterInfo.ByDiffContent.Should().BeFalse();
+        }
+
+        [Test]
+        public void FilterInfo_Apply_ByDiffContent()
+        {
+            FilterInfo filterInfo = new();
+            var filterLaunched = filterInfo.Apply(new RevisionFilter("diff", false, false, false, true));
+
+            filterLaunched.Should().BeTrue();
+            filterInfo.ByDiffContent.Should().BeTrue();
+            filterInfo.DiffContent.Should().Be("diff");
+
+            filterInfo.ByMessage.Should().BeFalse();
+            filterInfo.ByAuthor.Should().BeFalse();
+            filterInfo.ByCommitter.Should().BeFalse();
+        }
+
+        [Test]
+        public void FilterInfo_Apply_ByMessage_Then_By_DiffContent()
+        {
```

Comment:
A unit test should typically perform a single action and the assert the results. This test appears to perform several actions, and I'm not saying this is necessarily wrong, but some comments would help understanding what and why we are doing so.

Reply:
done.